### PR TITLE
Nept 1076 : Add Labels component for community profile

### PR DESCRIPTION
**Pull request by Medi4tor**
Implement Labels in community profile.

This label is used for the field 'community access'.
- There is no 'good' preview of it in atomium overview as it is something specific to community, and it's a special display for a field.

### CODE REVIEW

File: templates/field/field--list-boolean--node--full--community.tpl.php [hunk 1]
```diff
@@ -0,0 +1,12 @@
+<?php
+
+/**
+ * @file
+ * Contains template file.
+ */
+?>
+<span <?php print $attributes; ?>>
```
Comment (drupol) [inline templates/field/field--list-boolean--node--full--community.tpl.php]: Remove the extraneous empty space between <span and <?php.
The attribute processing system automatically adds it.

File: templates/field/field.component.inc [hunk 2]
```diff
@@ -51,3 +51,17 @@ function ec_europa_preprocess_field(&$variables, $hook) {
...
   }
 }
+
+/**
+ * Implements hook_preprocess_hook().
+ */
+function ec_europa_preprocess_field__list_boolean__node__full__community(&$variables, $hook) {
+  if ($variables['element']['#field_name'] == 'group_access') {
+    $node = (isset($variables['element']['#object'])) ? $variables['element']['#object'] : '';
+    if (is_object($node)) {
```
Comment (drupol) [inline templates/field/field.component.inc]: Maybe `if ($node = $variables['element']['#object'])` would be better ?

File: templates/labels/labels.component.inc [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+ * Contains component file.
+ */
+
+/**
+ * Implements hook_atomium_theme_hook().
+ */
+function ec_europa_atomium_theme_labels() {
+  return array(
+    'labels' => array(
+      'template' => 'field__list_boolean__node__full__community',
```
Comment (drupol) [inline templates/labels/labels.component.inc]: A component is a self contained (in it's own directory) set of files (.component.inc, .tpl.php, .js, .css) that lives inside the same directory.
Apparently, in this case, you're relying on a templace which is somewhere else, which is not good.

The component 'labels' should work out of the box, with it's own template.

File: templates/labels/labels.component.inc [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+}
+
+/**
+ * Implements hook_atomium_definition_form_hook().
+ */
+function ec_europa_atomium_definition_form_labels($form, &$form_state) {
+  $form['element1'] = array(
+    '#markup' => 'Example Open Status',
+  );
+  $form['#attributes']['class'][] = array('ecl-label', 'ecl-label--open');
```
Comment (drupol) [inline templates/labels/labels.component.inc]: These classes should be added in the `ec_europa_preprocess_labels()`.

To display the labels, you will use:

```php
$form['element1'] = array(
   '#theme' => 'labels',
   '#markup' => 'Example Open Status',
);
```

(_warning: I'm not sure that using #markup is allowed_)

File: templates/labels/labels.tpl.php [hunk 5]
```diff
@@ -0,0 +1,8 @@
+<?php
+
+/**
+ * @file
+ * Contains template file.
+ */
+?>
+<span <?php print $attributes; ?>><?php print $title; ?></span>
```
Comment (drupol) [inline templates/labels/labels.tpl.php]: The variable $title is not defined in `hook_atomium_theme_hook()`.

Please, have a look at the component that are already made and inspire your code from one of them.

File: templates/field/field--list-boolean--node--full--community.tpl.php (hunk 1 above)
Comment (Medi4tor) [reply templates/field/field--list-boolean--node--full--community.tpl.php]: :+1:

File: templates/field/field.component.inc (hunk 2 above)
Comment (Medi4tor) [reply templates/field/field.component.inc]: :+1:

File: templates/labels/labels.component.inc (hunk 3 above)
Comment (Medi4tor) [reply templates/labels/labels.component.inc]: Yes, indeed...
Here it's a 'special' component request for the theme ec_europa in the context of community profile.
I agree that is not generic enough.
I lost a lot of time on trying several solutions...

The template suggestion for this content is a field. so I add a specific  hook preprocess in field component. 
I will be glad to see how to achieve this in another way :)

File: templates/labels/labels.component.inc (hunk 4 above)
Comment (Medi4tor) [reply templates/labels/labels.component.inc]: Unfortunately, I was not able to make it work like this.

File: templates/labels/labels.tpl.php (hunk 5 above)
Comment (Medi4tor) [reply templates/labels/labels.tpl.php]: You're right , the file should be deleted. :+1 It's my git mistake .

**Comment by drupol** [comment]
:+1:

**Pull request merged**